cmd/kubeapps-apis/server/packages_test.go

```diff
@@ -0,0 +1,399 @@
+/*
+Copyright © 2021 VMware
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+    http://www.apache.org/licenses/LICENSE-2.0
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+package server
+
+import (
+	"context"
+	"testing"
+
+	"github.com/google/go-cmp/cmp"
+	"github.com/google/go-cmp/cmp/cmpopts"
+	corev1 "github.com/kubeapps/kubeapps/cmd/kubeapps-apis/gen/core/packages/v1alpha1"
+	plugins "github.com/kubeapps/kubeapps/cmd/kubeapps-apis/gen/core/plugins/v1alpha1"
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/status"
+)
+
+const (
+	globalPackagingNamespace = "kubeapps"
+)
+
+// TODO(agamez): currently, these tests are just calling the packaging core api functions,
+// however, I have to figure out how to fake the load of the plugins for each test case
```

Comment:
Yep, let's talk about it, but in overview, I think the only way to sanely test this view is to define a `makeTestPackagingPlugin()` that returns a struct that implements the interface by returning stored results. If the struct has public fields such as `AvailablePackageSummaries` etc., you can just set the responses you want. It's a valid plugin (implementing the same interface), so is valid for testing the contract of this core API. Of course the test plugins will be much faster than the helm or flux one, which is what you want for unit tests, but we can also simulate delays if required later.

Your test then can make two packaging plugins and assign the slice directly to the server struct. Chat more later.

Reply:
Mmmm, got it. Yep, it makes sense, Otherwise, mocking every plugin would become such a pain... and it doesn't really fit the unit test idea. We better add an integration test later.
Ok, I'm gonna work on that `makeTestPackagingPlugin`. Thanks for your comments!